server/routes/api/documents/documents.ts

```diff
@@ -343,7 +343,25 @@ router.post(
       const collectionIds = await user.collectionIds();
       where = {
         ...where,
-        collectionId: collectionIds,
+        [Op.or]: [
+          {
+            collectionId: {
+              [Op.in]: collectionIds,
+            },
+          },
+          can(user, "readDocument", user.team)
```

Comment:
Looking at this now, I'm not sure if an indirection through policy is really required here? Also the policy `readDocument` seems very restrictive to allow only admins. Instead, should we just get rid of the policy and directly use `isTeamAdmin` here? Makes it more explicit IMO.

cc: @tommoor

Reply:
We want to retrieve the workspace level archived documents the user has access to. For workspace level docs (regardless of their archived status), admins should be able to view everything, while other users should be able to access the docs owned by them. The condition statement here and `readDocument` policy allow us to achieve this.

re: policy, I think it gives us the flexibility to change who can access workspace level docs later.

Reply:
Fine with this, no big deal.